### PR TITLE
[Merged by Bors] - api: v2alpha1: ATX improvements and sort order filter

### PR DESCRIPTION
**Pull request by kacpersaw**
* Add NumUnits to Activation message
* Change name of NodeId filter to SmesherId.
* Modify Id to allow multiple entries.
* Remove signature and previous_atx from Activation message
* Update stream filters to match list filters
* Add sort_order field to tx, layer and reward List requests

### CODE REVIEW

File: api/grpcserver/v2alpha1/activation.go [hunk 1]
```diff
@@ -245,17 +246,17 @@ func toAtxOperations(filter *spacemeshv2alpha1.ActivationRequest) (builder.Opera
 	if filter == nil {
 		return ops, nil
 	}
-	if filter.NodeId != nil {
+	if filter.SmesherId != nil {
```
Comment (poszu) [inline api/grpcserver/v2alpha1/activation.go]: Consider 
```suggestion
	if len(filter.SmesherId) > 0 {
```

btw, the field could have a plural name to make it clearer it's a list.

File: api/grpcserver/v2alpha1/activation.go [hunk 2]
```diff
@@ -312,20 +313,28 @@ type atxsMatcher struct {
...
-		if t.SmesherID != nodeId {
+	if len(m.SmesherId) > 0 {
+		found := false
+		for _, id := range m.SmesherId {
+			nodeId := types.BytesToNodeID(id)
+			if t.SmesherID == nodeId {
+				found = true
+			}
+		}
+		if !found {
```
Comment (poszu) [inline api/grpcserver/v2alpha1/activation.go]: How about
```suggestion
		idx := slices.IndexFunc(m.SmesherId, func(id []byte) bool { return bytes.Equal(id, t.SmesherID.Bytes()) })
		if idx == -1 {
```

File: api/grpcserver/v2alpha1/activation.go [hunk 3]
```diff
@@ -312,20 +313,28 @@ type atxsMatcher struct {
...
-
-		if t.ID() != atxId {
+		found := false
+		for _, id := range m.Id {
+			atxId := types.BytesToATXID(id)
+			if t.ID() == atxId {
+				found = true
+			}
+		}
+		if !found {
```
Comment (poszu) [inline api/grpcserver/v2alpha1/activation.go]: Ditto:
```suggestion
		idx := slices.IndexFunc(m.Id, func(id []byte) bool { return bytes.Equal(id, t.ID().Bytes()) })
		if idx == -1 {
```

**Review by poszu — APPROVED** [review]

**Comment by kacpersaw** [comment]
bors merge

**Pull request closed without merging**